mlflow/models/utils.py

```diff
@@ -198,7 +203,8 @@ def _handle_dataframe_input(input_ex):
                 "type": example_type,
             }
         else:
-            self.data = _handle_dataframe_input(input_example)
+            self._inference_data = _coerce_to_pandas_df(input_example)
+            self.data = _handle_dataframe_input(self._inference_data)
```

Comment:
Do we need two attributes? Can we just use one?

Reply:
I considered doing this. It didn't work because right now `self.data` stores a dictionary, such as
```
{
    "data": _handle_ndarray_nans(x.data).tolist(),
    "indices": x.indices.tolist(),
    "indptr": x.indptr.tolist(),
    "shape": list(x.shape),
}
```
and the `_Example` class uses this dictionary to serialize the input example into JSON. `self._data` does not store the data in its numpy, sparse matrix, or pandas dataframe form (used for PyFunc inference). I added the `self._inference_data` property to store the data in that form.

Reply:
Makes sense! Is `_inference_data` always a pandas dataframe?

Reply:
Looks like it's not.

Reply:
Yeah, it is not always a pandas dataframe, but can also be a numpy and sparse matrix, which makes sense to me because some pyfunc-wrapped models should be called with those data types.

Reply:
Dumb question, should we rename `_coerce_to_pandas_df` to `_prep_for_inference` or something if it doesn't always return a pandas DF?